plugins/cumulus_frontend/web_client/js/views/body/ComputeResourcePanel.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * This view is in charge of the Compute Resources panel.
+ * The panel is backed by a collection of compute resource models.
+ *
+ * This panel is responsible for the following:
+ *   Viewing (and keeping up to date) a list of compute resources
+ *   Adding compute resources
+ *   Allowing a user to view a cluster's details
+ *   Terminating a cluster
+ *   Removing a cluster
+ *
+ * In the need for keeping the panel list up to date, 2 main types of event
+ * handling need to happen:
+ * 1) When the collection changes (via reset or remove) this view needs
+ * to be re-rendered but ALSO listen to changes in status on any individual
+ * model in the collection as this effects what type of icon would be displayed.
+ * 2) When the Girder event stream triggers a g:event.cluster.status, this indicates
+ * that one of the cluster models has changed status - and as such it gets set on the
+ * appropriate model. This action triggers a re-render through item 1.
+ **/
+minerva.views.ComputeResourcePanel = minerva.views.Panel.extend({
+    events: {
+        'click .m-add-computeresource': 'addComputeResourceDialog',
+        'click .m-cluster-details': function (e) {
+            var resource = this.collection.get($(e.currentTarget).attr('m-resource-id'));
+
+            new minerva.views.ComputeResourceDetailWidget({
+                el: $('#g-dialog-container'),
+                model: resource,
+                parentView: this
+            }).render(true);
+        },
```

Comment:
It looks like what you're suggesting (mostly) tends to be the convention Minerva follows. I'll switch this and the other views I added to use that convention.